src/Symfony/Component/DependencyInjection/Container.php

```diff
@@ -469,7 +469,9 @@ protected function getEnv($name)
             return $this->envCache[$name];
         }
         if (!$this->has($id = 'container.env_var_processors_locator')) {
-            $this->set($id, new ServiceLocator(array()));
+            // We don't call set() to prevent an infinite loop:
+            // set() calls getRemovedIds(), getRemovedIds() calls getEnv() in some cases
+            $this->services[$id] = new ServiceLocator(array());
```

Comment:
assigning directly is not good, as it bypasses a bunch of validation and processing (you don't normalize the id for instance)

Reply:
Here it's a very special case. The id is known and is already normalized and validated. We may add a third parameter to `set` to prevent the circular env resolution, but it looks overkill for this very special case.